service-workers/service-worker/waiting.https.html

```diff
@@ -5,36 +5,40 @@
 <script src="resources/test-helpers.sub.js"></script>
 <body>
 <script>
-promise_test(t => {
-    var script = 'resources/empty-worker.js';
-    var scope = 'resources/blank.html';
-    var frame;
-    var registration;
 
-    return service_worker_unregister(t, scope)
-      .then(() => { return with_iframe(scope); })
-      .then(f => {
-          frame = f;
-          return navigator.serviceWorker.register(script, {scope: scope});
-        })
-      .then(r => {
-          registration = r;
-          add_result_callback(() => { registration.unregister(); });
-          return wait_for_state(t, r.installing, 'installed');
-        })
-      .then(() => {
-          const controller =
-            frame.contentWindow.navigator.serviceWorker.controller;
-          assert_equals(controller, null, '.controller');
+const SCRIPT = 'resources/empty-worker.js';
+const SCOPE = 'resources/blank.html';
+let registration;
 
-          assert_equals(registration.active, null, '.active');
-          assert_equals(registration.waiting.state, 'installed', '.waiting');
-          assert_equals(registration.installing, null, '.installing');
-        })
-      .then(() => {
-          frame.remove();
-          return registration.unregister();
-        });
-  }, 'waiting is set after installation');
+promise_test(async t => {
+    let frame;
+    await service_worker_unregister(t, SCOPE);
+    frame = await with_iframe(SCOPE);
+    registration =
+        await navigator.serviceWorker.register(SCRIPT, {scope: SCOPE});
+    await wait_for_state(t, registration.installing, 'installed');
+    const controller =
+        await frame.contentWindow.navigator.serviceWorker.controller;
+
+    assert_equals(controller, null, '.controller');
+    assert_equals(registration.active, null, '.active');
+    assert_equals(registration.waiting.state, 'installed', '.waiting');
+    assert_equals(registration.installing, null, '.installing');
+    t.add_cleanup(() => frame.remove());
```

Comment:
It's better to remove the frame before calling unregister, to help the agent clean up.